src/main.cpp

```diff
@@ -36,7 +36,10 @@ int main(int argc, char** argv) {
     setenv("XDG_BACKEND", "wayland", 1);
     setenv("_JAVA_AWT_WM_NONREPARENTING", "1", 1);
     setenv("MOZ_ENABLE_WAYLAND", "1", 1);
-    setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
+
+    // Set XDG_CURRENT_DESKTOP if it was not set for us
+    if (!getenv("XDG_CURRENT_DESKTOP"))
+        setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
```

Comment:
you should also add a flag to not `unsetenv` it on exit if it wasn't overwritten

Reply:
Does this look right?
```
    // Set XDG_CURRENT_DESKTOP if it was not set for us
    bool xdgcdExisted = False;
    if (!getenv("XDG_CURRENT_DESKTOP")) {
        setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
    } else {
        xdgcdExisted = True;
    }
```
I'm afraid, connecting this to `CCompositor::cleanEnvironment()` might be beyond my current knowledge of C++ which is practically nonexistent.

Reply:
Maybe instead of storing `xdgcdExisted` it will be simpler to unset  XDG_CURRENT_DESKTOP only if it is `Hyprland`?

Reply:
If it came from desktop entry, this would be ambiguous.

Reply:
I'll do it myself then

Reply:
Thanks!